src/Servers/IIS/IIS/src/Core/IISHttpContext.IO.cs

```diff
@@ -159,6 +159,11 @@ private async Task WriteBody(bool flush = false)
         {
             while (true)
             {
+                if (_bodyOutput.Aborted)
+                {
+                    break;
```

Comment:
Not really sure this break here is ok, as I'm not really sure what the expectations are before/after this loop, is this what you had in mind @Tratcher ?

Reply:
We have tests explicitly checking that Writes after abort no-op, so I don't think we want to throw exceptions right?

Reply:
Right, Write shouldn't throw.

I think this check needs to be after ReadAsync or you'd miss an Abort that happened while waiting for data.

Reply:
Hmm, there might be an issue if AdvanceTo isn't called? Moving this check inside the Try block would cover that with the finally.

Reply:
Are we always calling Complete on the pipe? If we don't there's a memory leak.